README.md

```diff
@@ -1,8 +1,194 @@
+```
+## Private docker images ##
+
+At the moment, we don't want to give molecule access to private docker hub credentials for private docker images (e.g. shadow-teleop). That is why, in every playbook.yml inside the test scenarios in the molecule_docker folder, we override the image with image="shadowrobot/dexterous-hand" for any teleop-related test scenario. When we actually deploy Aurora, the user will be asked to fill in their private Docker hub credentials.
+
+## Testing with molecule_ec2 ##
+
+Once you have written your code for aurora in your branch, and tested it locally with molecule_docker, you can test it with AWS EC2 (initiated from local), by following the steps here:
+
+## Credentials ##
+
+1. Ask the system adminstrator for your AWS access key and secret access key. Then, in the docker container terminal, type:
+
+```
+aws configure
+```
+2. Paste the access key and the secret access key
+
+3. Default region name must be: eu-west-2
+
+4. Press enter on the Default format
+
+
+Then continue testing with molecule_ec2:
+
+1. In the docker container terminal go to /ansible/playbooks/molecule_ec2 folder
+
+```
+cd /home/user/aurora/ansible/playbooks/molecule_ec2
+```
+
+2. Run the following command to execute all AWS EC2 molecule tests in AWS (but you can see local debug logs):
+
+```
+molecule --debug test --all
+```
+
+3. Fix any errors
+
+4. If you want to run a specific test case (the AWS EC2 scenarios are in ansible/playbooks/molecule_ec2/molecule folder), use:
+
+```
+molecule --debug test -s name_of_your_scenario
+```
+5. Often it is useful to run molecule in stages (create, converge, verify, login (if necessary), and finally destory) for better debugging (so you can inspect every stage yourself). See [this](https://molecule.readthedocs.io/en/stable/usage.html) page, and do, for example:
+```
+molecule --debug create -s name_of_your_scenario
+molecule --debug converge -s name_of_your_scenario
+molecule --debug verify -s name_of_your_scenario
+molecule --degub login -s name_of_your_scenario
+molecule --debug destroy -s name_of_your_scenario
+```
+## Automatic tests ##
+
+The buildspec.yml file in the root of the project defines what AWS CodeBuild should run when a PR is created or updated or when a daily build runs. It is configured to run all tests in /ansible/playbooks/molecule_ec2 folder
+
+## Test creation ##
+
+Create test scenarios for both docker in ansible/playbooks/molecule_docker/molecule folder and for AWS EC2 in ansible/playbooks/molecule_ec2/molecule folder. For additional molecule_docker tests, copy the folder structure from other tests and modify the python .py file in tests folder.For additional molecule_ec2 tests, copy the folder structure of another EC2 test and modify the molecule.yml file inside. The EC2 tests just run the same tests as the Docker tests, but they do it in AWS EC2, using virtual machines, not Docker.
+
+## Testing or deploying on real hardware ##
```

Comment:
I would move this before tests section as its logical to first create the task and then make tests for it

Reply:
agree

Reply:
resolved